### ISSUE TITLE
IfcRelSpaceBoundary RelatingSpace GUID

### ISSUE THREAD

**Issue by jdimyadi** [xBimTeam/XbimEssentials#104]
I'm trying to get the GUID of the IfcSpace related to IfcRelSpaceBoundary via RelatingSpace. I can get the object, but cannot access the GUID of the object. Can anyone help please?

**Comment by CBenghi** [comment]
So you get a valid IfcSpace instance but its GlobalId field is null, is this the case?
Can you read the entitylabel?
Have you looked in the ifc file to check that the property is filled?
To be sure you can open the file with Xplorer, select the space and look at the properties there.
Let us know.

**Comment by jdimyadi** [comment]
What I got was  "Xbim.Ifc2x3.ProductExtension.IfcSpace", which was returned by RelatingSpace, but I want the GUID of the relating space.

Here's my code:
var relSpaceBoundaries = model.Instances.OfType<IIfcRelSpaceBoundary>();
foreach (var relSpaceBoundary in relSpaceBoundaries)
    {
        {
            var IfcRelSpaceBoundaryGuid = relSpaceBoundary.GlobalId
            var relatingSpaceGuid = relSpaceBoundary.RelatingSpace_.GlobalId_
        }
}

NOTE: I wanted to specify GlobalId as above, but the RelatingSpace does not have GlobalId property.
Would it be possible to concat Model.Instances.OfType<IIfcSpace>() ?

**Comment by jdimyadi** [comment]
In the code above "_.GlobalId_" was supposed to be in Italic.

The intention was "relSpaceBoundary.RelatingSpace.GlobalId" to obtain the GUID of the space object related to the IfcRelSpaceBoundary object

**Comment by martin1cerny** [comment]
This should work as it is. Can you open the file in a text editor to confirm the value is actually in there? If it is filled in it would be the first value on the row.

```
#152= IFCSPACE('3w0zWKm7n8SB1qbfwUzt0U',#41,'1 - Living room',$,$,#122,#148,'Living room',.ELEMENT.,.INTERNAL.,$);
                ^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by jdimyadi** [comment]
Yes, each IfcSpace object has a GUID. The issue I'm reporting is that RelatingSpace has NO property called GlobalId. So, I cannot access the GUID of the related space from IfcRelSpaceBoundary object.

**Comment by CBenghi** [comment]
@jdimyadi,
can you share a model where this happens (privately, if needed)?
Claudio

**Comment by jdimyadi** [comment]
@CBenghi Happy to share the model, but firstly could you please tell me if IfcRelSpaceBoundary.RelatingSpace has a GlobalId property? If not, what is the correct method of getting the GlobalId of an IfcSpace object related to its spaceboundary please?

**Comment by CBenghi** [comment]
Hello @jdimyadi,
sure; IfcRelSpaceBoundary.RelatingSpace is of type IfcSpace and any IfcSpace should have the property GlobalId populated, because in the schema it's not marked as optional.
Apologies for any confusion, I assumed that Martin's question and your answer above assumed common understanding of this aspect of the schema.

If you wish to send just a minimal file we have a GUI to make the extraction easy (I can show you if interested).

Claudio

**Comment by jdimyadi** [comment]
Hi @Claudio
Here's a simple model that has space boundaries. I'm interested to get a list of spaces related to each wall that shares the same space boundaries. Ultimately, I would like to identify windows and doors related to a space. Yes, please if you could show me how to achieve this.

[Test3.ifc.zip](https://github.com/xBimTeam/XbimEssentials/files/1301631/Test3.ifc.zip)

**Comment by jdimyadi** [comment]
Oops. Sorry, I attached a wrong file. Should be this one.

[Test4.ifc.zip](https://github.com/xBimTeam/XbimEssentials/files/1301635/Test4.ifc.zip)

**Comment by martin1cerny** [comment]
`RelatingSpace` is a select type which is not a part of inheritance hierarchy so it doesn't appear to be `IfcRoot` with `GlobalId` property. However, both members of the select type are `IIfcRoot` so you can just cast it:

```cs
var relSpaceBoundaries = model.Instances.OfType<IIfcRelSpaceBoundary>();
foreach (var relSpaceBoundary in relSpaceBoundaries)
{
    {
        var IfcRelSpaceBoundaryGuid = relSpaceBoundary.GlobalId;
        var relatingSpaceGuid = (relSpaceBoundary.RelatingSpace as IIfcRoot)?.GlobalId;
    }
}
```

**Comment by jdimyadi** [comment]
Thank you, @martin1cerny
